"error: no member named 'tr1' in namespace 'std'" when building with java or python

Because there is no include-path for `boost` in command-line.
```
[ 99%] Building C object RenderSystems/GL/CMakeFiles/RenderSystem_GL.dir/src/glew.c.o
cd /tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/RenderSystems/GL && /usr/local/Homebrew/Library/Homebrew/shims/super/clang -DBOOST_ALL_NO_LIB -DGLEW_STATIC -DOGRE_GLPLUGIN_EXPORTS -DRenderSystem_GL_EXPORTS -DYY_NEVER_INTERACTIVE -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0 -I/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/RenderSystems/GL/include -I/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/OgreMain/include -I/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/include -I/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/OgreMain/include/OSX -I/usr/local/include -I/usr/local/include/freetype2 -I/Library/Frameworks/Cg.framework/Headers -F/Library/Frameworks -I/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9 -I/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/RenderSystems/GLSupport/include -I/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/RenderSystems/GLSupport/include/GLSL -I/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/RenderSystems/GL/src/GLSL/include -I/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/RenderSystems/GL/src/atifs/include -I/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/RenderSystems/GL/src/nvparse -I/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/OgreMain/include/Threading  -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -mmacosx-version-min=10.7 -fPIC   -pthread -DOGRE_GCC_VISIBILITY -fvisibility=hidden -fvisibility-inlines-hidden -o CMakeFiles/RenderSystem_GL.dir/src/glew.c.o   -c /tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/RenderSystems/GL/src/glew.c
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287646:47: error: no member named 'tr1' in namespace 'std'
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::Node * > > > result;
                                       ~~~~~~~^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287646:79: error: expected '(' for function-style cast or type construction
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::Node * > > > result;
                                                                  ~~~~~~~~~~~~^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287646:95: error: expected a type
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::Node * > > > result;
                                                                                              ^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287656:7: error: use of undeclared identifier 'result'
      result = (arg1)->getChildIterator();
      ^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287667:121: error: use of undeclared identifier 'result'
  resultobj = SWIG_NewPointerObj((new Ogre::Node::ChildNodeIterator(static_cast< const Ogre::Node::ChildNodeIterator& >(result))), SWIGTYPE_p_Ogre__MapIteratorT_std__tr1__unordered_mapT_Ogre__String_Ogre__Node_p_t_t, SWIG_POINTER_OWN |  0 );
                                                                                                                        ^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287679:52: error: no member named 'tr1' in namespace 'std'
  SwigValueWrapper< Ogre::ConstMapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::Node * > > > result;
                                            ~~~~~~~^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287679:84: error: expected '(' for function-style cast or type construction
  SwigValueWrapper< Ogre::ConstMapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::Node * > > > result;
                                                                       ~~~~~~~~~~~~^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287679:100: error: expected a type
  SwigValueWrapper< Ogre::ConstMapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::Node * > > > result;
                                                                                                   ^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287689:7: error: use of undeclared identifier 'result'
      result = ((Ogre::Node const *)arg1)->getChildIterator();
      ^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287700:131: error: use of undeclared identifier 'result'
  resultobj = SWIG_NewPointerObj((new Ogre::Node::ConstChildNodeIterator(static_cast< const Ogre::Node::ConstChildNodeIterator& >(result))), SWIGTYPE_p_Ogre__ConstMapIteratorT_std__tr1__unordered_mapT_Ogre__String_Ogre__Node_p_t_t, SWIG_POINTER_OWN |  0 );
                                                                                                                                  ^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291448:47: error: no member named 'tr1' in namespace 'std'
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::MovableObject * > > > result;
                                       ~~~~~~~^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291448:79: error: expected '(' for function-style cast or type construction
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::MovableObject * > > > result;
                                                                  ~~~~~~~~~~~~^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291448:104: error: expected a type
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::MovableObject * > > > result;
                                                                                                       ^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291458:7: error: use of undeclared identifier 'result'
      result = (arg1)->getAttachedObjectIterator();
      ^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291469:125: error: use of undeclared identifier 'result'
  resultobj = SWIG_NewPointerObj((new Ogre::SceneNode::ObjectIterator(static_cast< const Ogre::SceneNode::ObjectIterator& >(result))), SWIGTYPE_p_Ogre__MapIteratorT_std__tr1__unordered_mapT_Ogre__String_Ogre__MovableObject_p_t_t, SWIG_POINTER_OWN |  0 );
                                                                                                                            ^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291481:52: error: no member named 'tr1' in namespace 'std'
  SwigValueWrapper< Ogre::ConstMapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::MovableObject * > > > result;
                                            ~~~~~~~^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291481:84: error: expected '(' for function-style cast or type construction
  SwigValueWrapper< Ogre::ConstMapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::MovableObject * > > > result;
                                                                       ~~~~~~~~~~~~^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291481:109: error: expected a type
  SwigValueWrapper< Ogre::ConstMapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::MovableObject * > > > result;
                                                                                                            ^
/tmp/ogre-20171107-1582-t1s75t/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291491:7: error: use of undeclared identifier 'result'
      result = ((Ogre::SceneNode const *)arg1)->getAttachedObjectIterator();
      ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
```

Using "std" instead of `boost`, Error occurred.
```
[ 97%] Building CXX object RenderSystems/GL/CMakeFiles/RenderSystem_GL.dir/src/OgreGLPBRenderTexture.cpp.o
cd /tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/RenderSystems/GL && /usr/local/Homebrew/Library/Homebrew/shims/super/clang++  -DGLEW_STATIC -DOGRE_GLPLUGIN_EXPORTS -DRenderSystem_GL_EXPORTS -DYY_NEVER_INTERACTIVE -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0 -I/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/RenderSystems/GL/include -I/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/OgreMain/include -I/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/include -I/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/OgreMain/include/OSX -I/usr/local/include -I/usr/local/include/freetype2 -I/Library/Frameworks/Cg.framework/Headers -F/Library/Frameworks -I/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9 -I/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/RenderSystems/GLSupport/include -I/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/RenderSystems/GLSupport/include/GLSL -I/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/RenderSystems/GL/src/GLSL/include -I/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/RenderSystems/GL/src/atifs/include -I/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/RenderSystems/GL/src/nvparse -I/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/OgreMain/include/Threading  -std=c++11 -stdlib=libc++ -msse -Wall -Winit-self -Wcast-qual -Wwrite-strings -Wextra -Wundef -Wmissing-declarations -Wno-unused-parameter -Wshadow -Wno-missing-field-initializers -Wno-long-long -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -mmacosx-version-min=10.7 -fPIC   -pthread -DOGRE_GCC_VISIBILITY -fvisibility=hidden -fvisibility-inlines-hidden -o CMakeFiles/RenderSystem_GL.dir/src/OgreGLPBRenderTexture.cpp.o -c /tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/RenderSystems/GL/src/OgreGLPBRenderTexture.cpp
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287646:47: error: no member named 'tr1' in namespace 'std'
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::Node * > > > result;
                                       ~~~~~~~^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287646:79: error: expected '(' for function-style cast or type construction
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::Node * > > > result;
                                                                  ~~~~~~~~~~~~^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287646:95: error: expected a type
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::Node * > > > result;
                                                                                              ^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287656:7: error: use of undeclared identifier 'result'
      result = (arg1)->getChildIterator();
      ^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287667:121: error: use of undeclared identifier 'result'
  resultobj = SWIG_NewPointerObj((new Ogre::Node::ChildNodeIterator(static_cast< const Ogre::Node::ChildNodeIterator& >(result))), SWIGTYPE_p_Ogre__MapIteratorT_std__tr1__unordered_mapT_Ogre__String_Ogre__Node_p_t_t, SWIG_POINTER_OWN |  0 );
                                                                                                                        ^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287679:52: error: no member named 'tr1' in namespace 'std'
  SwigValueWrapper< Ogre::ConstMapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::Node * > > > result;
                                            ~~~~~~~^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287679:84: error: expected '(' for function-style cast or type construction
  SwigValueWrapper< Ogre::ConstMapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::Node * > > > result;
                                                                       ~~~~~~~~~~~~^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287679:100: error: expected a type
  SwigValueWrapper< Ogre::ConstMapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::Node * > > > result;
                                                                                                   ^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287689:7: error: use of undeclared identifier 'result'
      result = ((Ogre::Node const *)arg1)->getChildIterator();
      ^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:287700:131: error: use of undeclared identifier 'result'
  resultobj = SWIG_NewPointerObj((new Ogre::Node::ConstChildNodeIterator(static_cast< const Ogre::Node::ConstChildNodeIterator& >(result))), SWIGTYPE_p_Ogre__ConstMapIteratorT_std__tr1__unordered_mapT_Ogre__String_Ogre__Node_p_t_t, SWIG_POINTER_OWN |  0 );
                                                                                                                                  ^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291448:47: error: no member named 'tr1' in namespace 'std'
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::MovableObject * > > > result;
                                       ~~~~~~~^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291448:79: error: expected '(' for function-style cast or type construction
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::MovableObject * > > > result;
                                                                  ~~~~~~~~~~~~^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291448:104: error: expected a type
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::MovableObject * > > > result;
                                                                                                       ^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291458:7: error: use of undeclared identifier 'result'
      result = (arg1)->getAttachedObjectIterator();
      ^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291469:125: error: use of undeclared identifier 'result'
  resultobj = SWIG_NewPointerObj((new Ogre::SceneNode::ObjectIterator(static_cast< const Ogre::SceneNode::ObjectIterator& >(result))), SWIGTYPE_p_Ogre__MapIteratorT_std__tr1__unordered_mapT_Ogre__String_Ogre__MovableObject_p_t_t, SWIG_POINTER_OWN |  0 );
                                                                                                                            ^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291481:52: error: no member named 'tr1' in namespace 'std'
  SwigValueWrapper< Ogre::ConstMapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::MovableObject * > > > result;
                                            ~~~~~~~^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291481:84: error: expected '(' for function-style cast or type construction
  SwigValueWrapper< Ogre::ConstMapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::MovableObject * > > > result;
                                                                       ~~~~~~~~~~~~^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291481:109: error: expected a type
  SwigValueWrapper< Ogre::ConstMapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::MovableObject * > > > result;
                                                                                                            ^
/tmp/ogre-20171107-12648-1snph44/ogre-1.10.9/Components/Python/OgrePYTHON_wrap.cxx:291491:7: error: use of undeclared identifier 'result'
      result = ((Ogre::SceneNode const *)arg1)->getAttachedObjectIterator();
      ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
```
[cmake_without boost.log](https://github.com/OGRECave/ogre/files/1447092/cmake_without.boost.log)

```
$ ls /usr/include/c++/4.2.1/tr1
array			ctype.h			repeat.h
bind_iterate.h		cwchar			stdarg.h
bind_repeat.h		cwctype			stdbool.h
boost_shared_ptr.h	fenv.h			stdint.h
cctype			float.h			stdio.h
cfenv			functional		stdlib.h
cfloat			functional_hash.h	tgmath.h
cinttypes		functional_iterate.h	tuple
climits			hashtable		tuple_defs.h
cmath			hashtable_policy.h	tuple_iterate.h
common.h		inttypes.h		type_traits
complex			limits.h		type_traits_fwd.h
cstdarg			math.h			unordered_map
cstdbool		memory			unordered_set
cstdint			mu_iterate.h		utility
cstdio			random			wchar.h
cstdlib			random.tcc		wctype.h
ctgmath			ref_fwd.h
ctime			ref_wrap_iterate.h
```

Adding "-DCMAKE_CXX_FLAGS=-stdlib=libstdc++", The following error occurred.
```
cd /tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain && /usr/local/Homebrew/Library/Homebrew/shims/super/clang++  -DFREEIMAGE_LIB -DOgreMain_EXPORTS -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0 -I/tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain/include -I/tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/include -I/tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain/include/OSX -I/usr/local/include -I/usr/local/include/freetype2 -I/Library/Frameworks/Cg.framework/Headers -F/Library/Frameworks -I/tmp/ogre-20171107-16418-105pecy/ogre-1.10.9 -I/tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain/include/Threading  -stdlib=libstdc++ -msse -Wall -Winit-self -Wcast-qual -Wwrite-strings -Wextra -Wundef -Wmissing-declarations -Wno-unused-parameter -Wshadow -Wno-missing-field-initializers -Wno-long-long -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -mmacosx-version-min=10.7 -fPIC   -pthread -DOGRE_GCC_VISIBILITY -fvisibility=hidden -fvisibility-inlines-hidden -o CMakeFiles/OgreMain.dir/src/OgreAnimation.cpp.o -c /tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain/src/OgreAnimation.cpp
In file included from /tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain/src/OgreAlignedAllocator.cpp:28:
In file included from /tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain/include/OgreStableHeaders.h:45:
In file included from /tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain/include/OgreCamera.h:36:
In file included from /tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain/include/OgreFrustum.h:32:
In file included from /tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain/include/OgreMovableObject.h:36:
In file included from /tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain/include/OgreShadowCaster.h:32:
In file included from /tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain/include/OgreRenderable.h:34:
In file included from /tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain/include/OgreGpuProgram.h:33:
In file included from /tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain/include/OgreResource.h:33:
In file included from /tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain/include/OgreStringInterface.h:34:
In file included from /tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain/include/Threading/OgreThreadHeaders.h:40:
/tmp/ogre-20171107-16418-105pecy/ogre-1.10.9/OgreMain/include/Threading/OgreThreadHeadersSTD.h:30:10: fatal error: 'thread' file not found
#include <thread>
         ^~~~~~~~
```

Replaced in Components/Python/CMakeLists.txt as following,  Python components was given.
But errors occurred in building java components.
```
 set(PYTHON_SITE_PACKAGES lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/dist-packages/)

-set(CMAKE_SWIG_FLAGS -w401,314 -builtin)
+set(CMAKE_SWIG_FLAGS -w401,314 -builtin -D_LIBCPP_VERSION)

 macro(ogre_python_module target)
```
```
cd /tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/Components/Python && /usr/local/bin/swig -python -w401,314 -builtin -D_LIBCPP_VERSION -outdir /tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/Components/Python -c++ -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/OgreMain/include -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/include -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/OgreMain/include/OSX -I/usr/include -I/usr/local/include -I/usr/local/include/freetype2 -I/Library/Frameworks/Cg.framework/Headers -I/Library/Frameworks/Cg.framework -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9 -I/usr/include/python2.7 -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/Components/RTShaderSystem/include -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/Components/Overlay/include -o /tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/Components/Python/OgreOverlayPYTHON_wrap.cxx /tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/Components/Overlay/include/OgreOverlay.i
```

```
cd /tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/Components/Java && /usr/local/Homebrew/Library/Homebrew/shims/super/clang++  -DOgreJNI_EXPORTS -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0 -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/OgreMain/include -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/include -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/OgreMain/include/OSX -I/usr/local/include -I/usr/local/include/freetype2 -I/Library/Frameworks/Cg.framework/Headers -F/Library/Frameworks -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9 -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk/System/Library/Frameworks/JavaVM.framework/Headers -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/Components/RTShaderSystem/include -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/Components/Overlay/include -I/usr/local/include/SDL2 -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/Components/Bites/include  -std=c++11 -stdlib=libc++ -msse -Wall -Winit-self -Wcast-qual -Wwrite-strings -Wextra -Wundef -Wmissing-declarations -Wno-unused-parameter -Wshadow -Wno-missing-field-initializers -Wno-long-long -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -mmacosx-version-min=10.7 -fPIC   -pthread -Wno-strict-aliasing -Wno-cast-qual -Wno-shadow -Wno-missing-declarations -o CMakeFiles/OgreJNI.dir/__/__/java/src/org/Ogre/OgreBitesJAVA_wrap.cxx.o -c /tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre/OgreBitesJAVA_wrap.cxx
/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre/OgreJAVA_wrap.cxx:181552:47: error: no member named 'tr1' in namespace 'std'
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::Node * > > > result;
                                       ~~~~~~~^
/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre/OgreJAVA_wrap.cxx:181552:79: error: expected '(' for function-style cast or type construction
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::Node * > > > result;
                                                                  ~~~~~~~~~~~~^
/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre/OgreJAVA_wrap.cxx:181552:95: error: expected a type
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::Node * > > > result;
                                                                                              ^
/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre/OgreJAVA_wrap.cxx:181560:7: error: use of undeclared identifier 'result'; did you mean 'jresult'?
      result = (arg1)->getChildIterator();
      ^~~~~~
      jresult
/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre/OgreJAVA_wrap.cxx:181550:9: note: 'jresult' declared here
  jlong jresult = 0 ;
        ^
/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre/OgreJAVA_wrap.cxx:181560:24: error: assigning to 'jlong' (aka 'long long') from incompatible type 'ChildNodeIterator' (aka 'MapIterator<unordered_map<basic_string<char, char_traits<char>, allocator<char> >, Ogre::Node *> >')
      result = (arg1)->getChildIterator();
               ~~~~~~~~^~~~~~~~~~~~~~~~~~
/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre/OgreJAVA_wrap.cxx:181573:122: error: use of undeclared identifier 'result'
  *(Ogre::Node::ChildNodeIterator **)&jresult = new Ogre::Node::ChildNodeIterator((const Ogre::Node::ChildNodeIterator &)result); 
                                                                                                                         ^
/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre/OgreJAVA_wrap.cxx:183750:47: error: no member named 'tr1' in namespace 'std'
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::MovableObject * > > > result;
                                       ~~~~~~~^
/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre/OgreJAVA_wrap.cxx:183750:79: error: expected '(' for function-style cast or type construction
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::MovableObject * > > > result;
                                                                  ~~~~~~~~~~~~^
/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre/OgreJAVA_wrap.cxx:183750:104: error: expected a type
  SwigValueWrapper< Ogre::MapIterator< ::std::tr1::unordered_map< Ogre::String,Ogre::MovableObject * > > > result;
                                                                                                       ^
/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre/OgreJAVA_wrap.cxx:183758:7: error: use of undeclared identifier 'result'; did you mean 'jresult'?
      result = (arg1)->getAttachedObjectIterator();
      ^~~~~~
      jresult
/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre/OgreJAVA_wrap.cxx:183748:9: note: 'jresult' declared here
  jlong jresult = 0 ;
        ^
/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre/OgreJAVA_wrap.cxx:183758:24: error: assigning to 'jlong' (aka 'long long') from incompatible type 'ObjectIterator' (aka 'MapIterator<unordered_map<basic_string<char, char_traits<char>, allocator<char> >, Ogre::MovableObject *> >')
      result = (arg1)->getAttachedObjectIterator();
               ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre/OgreJAVA_wrap.cxx:183771:128: error: use of undeclared identifier 'result'
  *(Ogre::SceneNode::ObjectIterator **)&jresult = new Ogre::SceneNode::ObjectIterator((const Ogre::SceneNode::ObjectIterator &)result); 
                                                                                                                               ^
12 errors generated.
make[2]: *** [Components/Java/CMakeFiles/OgreJNI.dir/__/__/java/src/org/Ogre/OgreJAVA_wrap.cxx.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [Components/Java/CMakeFiles/OgreJNI.dir/all] Error 2

```
```
cd /tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/Components/Java && /usr/local/bin/swig -java -w401,314 -package org.Ogre -DOGRE_BITES_HAVE_SDL=1 -outdir /tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre -c++ -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/OgreMain/include -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/include -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/OgreMain/include/OSX -I/usr/include -I/usr/local/include -I/usr/local/include/freetype2 -I/Library/Frameworks/Cg.framework/Headers -I/Library/Frameworks/Cg.framework -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9 -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk/System/Library/Frameworks/JavaVM.framework/Headers -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/Components/RTShaderSystem/include -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/Components/Overlay/include -I/usr/local/include/SDL2 -I/tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/Components/Bites/include -o /tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/java/src/org/Ogre/OgreBitesJAVA_wrap.cxx /tmp/ogre-20171107-21909-1f7j1pl/ogre-1.10.9/Components/Bites/include/OgreBites.i
```

--- a/Components/Java/CMakeLists.txt
+++ b/Components/Java/CMakeLists.txt
@@ -26,6 +26,12 @@ else()
 endif()
 
 set(CMAKE_SWIG_FLAGS -w401,314 -package org.Ogre)
+
+if (APPLE AND NOT ANDROID)
+	# we are using libc++ on OSX
+	list(APPEND CMAKE_SWIG_FLAGS -D_LIBCPP_VERSION)
+endif()
+
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${OGRE_BINARY_DIR}/java/libs")
 set(CMAKE_SWIG_OUTDIR "${OGRE_BINARY_DIR}/java/src/org/Ogre")
 

--- a/Components/Python/CMakeLists.txt
+++ b/Components/Python/CMakeLists.txt
@@ -13,6 +13,11 @@ set(PYTHON_SITE_PACKAGES lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINO
 
 set(CMAKE_SWIG_FLAGS -w401,314 -builtin)
 
+if (APPLE AND NOT ANDROID)
+    # we are using libc++ on OSX
+    list(APPEND CMAKE_SWIG_FLAGS -D_LIBCPP_VERSION)
+endif()
+
 macro(ogre_python_module target)
     set_target_properties(${SWIG_MODULE_${target}_REAL_NAME} PROPERTIES 
                         DEBUG_POSTFIX ""

--- a/OgreMain/include/OgrePrerequisites.h
+++ b/OgreMain/include/OgrePrerequisites.h
@@ -54,7 +54,7 @@ namespace Ogre {
     // Define ogre version
     #define OGRE_VERSION_MAJOR 1
     #define OGRE_VERSION_MINOR 10
-    #define OGRE_VERSION_PATCH 9
+    #define OGRE_VERSION_PATCH 10
     #define OGRE_VERSION_SUFFIX ""
     #define OGRE_VERSION_NAME "Xalafu"
 

--- a/OgreMain/src/OgreMesh.cpp
+++ b/OgreMain/src/OgreMesh.cpp
@@ -123,6 +123,7 @@ namespace Ogre {
         }
         SubMeshList::iterator i = mSubMeshList.begin();
         std::advance(i, index);
+        OGRE_DELETE *i;
         mSubMeshList.erase(i);
         
         // Fix up any name/index entries

--- a/RenderSystems/GL3Plus/include/GLSL/OgreGLSLProgram.h
+++ b/RenderSystems/GL3Plus/include/GLSL/OgreGLSLProgram.h
@@ -66,15 +66,6 @@ namespace Ogre {
     class _OgreGL3PlusExport GLSLProgram : public GLSLProgramCommon
     {
     public:
-        /// Constructor should only be used by GLSLMonolithicProgramManager and GLSLSeparableProgramManager
-        GLSLProgram(GLSLShader* vertexProgram,
-                    GLSLShader* hullProgram,
-                    GLSLShader* domainProgram,
-                    GLSLShader* geometryProgram,
-                    GLSLShader* fragmentProgram,
-                    GLSLShader* computeProgram);
-        virtual ~GLSLProgram(void);
-
         void bindFixedAttributes(GLuint program);
 
         GLSLShader* getVertexShader() const { return static_cast<GLSLShader*>(mVertexShader); }
@@ -86,6 +77,14 @@ namespace Ogre {
 
         void setTransformFeedbackVaryings(const std::vector<String>& nameStrings);
     protected:
+        /// Constructor should only be used by GLSLMonolithicProgramManager and GLSLSeparableProgramManager
+        GLSLProgram(GLSLShader* vertexProgram,
+                    GLSLShader* hullProgram,
+                    GLSLShader* domainProgram,
+                    GLSLShader* geometryProgram,
+                    GLSLShader* fragmentProgram,
+                    GLSLShader* computeProgram);
+
         /// Container of atomic counter uniform references that are active in the program object
         GLAtomicCounterReferenceList mGLAtomicCounterReferences;
         /// Map of shared parameter blocks to uniform buffer references

--- a/RenderSystems/GL3Plus/src/GLSL/OgreGLSLProgram.cpp
+++ b/RenderSystems/GL3Plus/src/GLSL/OgreGLSLProgram.cpp
@@ -50,13 +50,6 @@ namespace Ogre {
     {
     }
 
-
-    GLSLProgram::~GLSLProgram(void)
-    {
-        OGRE_CHECK_GL_ERROR(glDeleteProgram(mGLProgramHandle));
-    }
-
-
     Ogre::String GLSLProgram::getCombinedName()
     {
         String name;

--- a/RenderSystems/GLES2/src/GLSLES/include/OgreGLSLESProgramCommon.h
+++ b/RenderSystems/GLES2/src/GLSLES/include/OgreGLSLESProgramCommon.h
@@ -58,10 +58,10 @@ namespace Ogre {
 #endif
 
         static void _writeToCache(const String& name, GLuint programHandle);
-    public:
+
         /// Constructor should only be used by GLSLESLinkProgramManager and GLSLESProgramPipelineManager
         GLSLESProgramCommon(GLSLESProgram* vertexProgram, GLSLESProgram* fragmentProgram);
-        virtual ~GLSLESProgramCommon(void);
+    public:
 
         void bindFixedAttributes(GLuint program);
 

--- a/RenderSystems/GLES2/src/GLSLES/src/OgreGLSLESProgramCommon.cpp
+++ b/RenderSystems/GLES2/src/GLSLES/src/OgreGLSLESProgramCommon.cpp
@@ -42,13 +42,7 @@ namespace Ogre {
     , mFragmentProgram(fragmentProgram)
     {
     }
-    
-    //-----------------------------------------------------------------------
-    GLSLESProgramCommon::~GLSLESProgramCommon(void)
-    {
-        OGRE_CHECK_GL_ERROR(glDeleteProgram(mGLProgramHandle));
-    }
-    
+
     //-----------------------------------------------------------------------
     Ogre::String GLSLESProgramCommon::getCombinedName()
     {

